autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -247,13 +247,90 @@ def send_warnings_to_log(message, category, filename, lineno,
 
 
 class AbstractEvaluator(object):
+    """
+    This method defines the interface that pipeline evaluators should follow, when
+    interacting with SMAC through ExecuteTaFuncWithQueue.
+
+    An evaluator is an object that:
+        + constructs a pipeline (be it a classification or regression estimator) for a given
+          pipeline_config and run settings (budget, seed)
+        + Fits and train this pipeline (TrainEvaluator) or test a given
+          configuration (TestEvaluator)
+
+    The provided configuration determines the type of pipeline created:
+        int: A dummy classifier/dummy regressor is created. This estimator serves
+             as a baseline model to ignore all models that perform worse than this
+             dummy estimator. Also, in the worst case scenario, this is the final
+             estimator created (for instance, in case not enough memory was allocated
+             and all configurations crashed).
+        str: A traditional pipeline is created, as the configuration will contain a
+             string defining the estimator to use. For example 'RandomForest'.
+        Configuration: A pipeline object matching this configuration object is created. This
+             is the case of neural architecture search, where different backbones
+             and head can be passed in the form of a configuration object.
+
+    Attributes:
+        backend (Backend):
+            An object to interface with the disk storage. In particular, allows to
+            access the train and test datasets
+        queue (Queue):
+            Each worker available will instantiate an evaluator, and after completion,
+            it will return the evaluation result via a multiprocessing queue
+        metric (autoPyTorchMetric):
+            A scorer object that is able to evaluate how good a pipeline was fit. It
+            is a wrapper on top of the actual score method (a wrapper on top of scikit
+            lean accuracy for example) that formats the predictions accordingly.
+        budget: (float):
```

Comment:
I think this function is agnostic to the budget magnitude. In case we decide to change the budget as desired. For example, the budget magnitude shifts with successive halving.